rts/lra/jaxrs/src/main/java/io/narayana/lra/filter/ServerLRAFilter.java

```diff
@@ -460,46 +575,149 @@ private boolean isJaxRsCancel(ContainerRequestContext requestContext, ContainerR
         return false;
     }
 
-    private URI startLRA(URI parentLRA, Method method, Long timeout) {
-        // timeout should already have been converted to milliseconds
-        String clientId = method.getDeclaringClass().getName() + "#" + method.getName();
+    // the request filter may perform multiple and in failure scenarios the LRA may be left in an ambiguous state:
+    // the following structure is used to track progress so that such failures can be reported in the response
+    // filter processing
+    private enum ProgressStep {
+        Left ("leave succeeded"),
+        LeaveFailed("leave failed"),
+        Started("start ok"),
+        StartFailed("start failed"),
+        Joined("join succeeded"),
+        JoinFailed("join failed"),
+        Ended("end ok"),
```

Comment:
Should not be `"end suceeded"`?

Reply:
Maybe, or "join ok" would work too from a consistency point of view.